File: bddtests/src/main/kotlin/mocking/tpp/models/SiteDetails.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlElement

@XmlAccessorType(XmlAccessType.FIELD)
data class SiteDetails(
        @XmlAttribute var unitName: String = "default unit name",
        @field:XmlElement(name = "Address") var address: TppAddress = TppAddress()
)

File: xamarinintegrationtests/NHSOnline.HttpMocks/Tpp/Models/SiteDetails.cs
using System.Xml.Serialization;

namespace NHSOnline.HttpMocks.Tpp.Models
{
    public sealed class SiteDetails
    {
        [XmlAttribute("unitName")]
        public string? UnitName { get; set; }

        public Address? Address { get; set; }
    }
}